src/caustic/cosmology.py

```diff
@@ -34,127 +36,102 @@
 )
 
 
-class Cosmology(Base):
-    def __init__(
-        self,
-        h0,
-        device: torch.device = torch.device("cpu"),
-        dtype: torch.dtype = torch.float32,
-    ):
-        super().__init__(device, dtype)
-        self.h0 = h0
+class Cosmology(Parametrized):
+    """
+    Units:
+        - Distance: Mpc
+        - Mass: solMass
+    """
+
+    def __init__(self, name: str):
+        super().__init__(name)
 
     @abstractmethod
-    def rho_cr(self, z) -> torch.Tensor:
+    def rho_cr(self, z: Tensor, x: dict[str, Any] = defaultdict(list)) -> Tensor:
```

Comment:
Good point. I did this because I wanted the default assumption to be that all parameters are static, so you don't need to pass anything in for `x`. I'd suggest we make `x` default to `None` everywhere and update `unpack` here
https://github.com/Ciela-Institute/caustic/blob/b394cb6833323791a87d5b77c15b82f6090a4d60/src/caustic/parametrized.py#L236
to instead be
```python
my_x = x[self.name] if isinstance(x, dict) else defaultdict(list)
```